ghmirror/core/mirror_requests.py

```diff
@@ -210,16 +180,16 @@ def online_request(method, url, auth, data=None, url_params=None):
     )
 
     if resp.status_code == 304:
-        return _handle_not_changed(
-            cached_response,
-            per_page_elements,
-            headers,
-            method,
-            url,
-            parameters,
-            cache,
-            cache_key,
-        )
+        # Pagination data can change withouth chaging the ETAG or
+        # the last-modified header. Returning the updated pagination data
+        # ensures all data is fetched in all cases.
+        if cached_response.links != resp.links:
+            cached_response.headers["Link"] = resp.headers["Link"]
```

Comment:
Will `resp.headers` always contain `Link`? If not this may throw a `KeyError` exception, may worth adding a log here. Any other headers need update too?

Reply:
* 🤦
* `Link` might be removed if pagination is not needed anymore -> Fixed. 
* There are other headers that gets updated like rate limit ones, but those does not affect how the responses are managed like the pagination data. 
* I also Improved the test